client/my-sites/checkout/checkout/pay-button.jsx

```diff
@@ -118,11 +119,54 @@ export class PayButton extends React.Component {
 
 		return this.props.translate( 'Pay now', { context: 'Pay button on /checkout' } );
 	};
+	beforeSubmitTextVariant = () => {
+		const cart = this.props.cart;
+
+		if ( this.props.beforeSubmitTextVariant ) {
+			return this.props.beforeSubmitTextVariant;
+		}
+
+		if ( hasOnlyFreeTrial( cart ) ) {
+			return this.props.translate( 'Start %(days)s Day Free Trial', {
+				args: { days: '14' },
+				context: 'Pay button for free trials on /checkout',
+			} );
+		}
+
+		if ( cart.total_cost_display ) {
+			if ( isPaidForFullyInCredits( cart ) ) {
+				if ( hasRenewalItem( this.props.cart ) ) {
+					return this.props.translate( 'Complete subscription purchase with Credits', {
+						context: 'Renew button on /checkout',
+					} );
+				}
+
+				return this.props.translate( 'Complete purchase with Credits', {
```

Comment:
:information_source: This **context** is really long. Are you sure you don't want to use a translator comment instead?